src/app/models/query.spec.ts

```diff
@@ -128,8 +138,285 @@ describe('function: parseQueryToString', () => {
 
 			const expectedResult = 'foo near:bar since:2017-06-22_04:25 until:2017-05-27_06:25';
 
-			const result: string = parseQueryToString(query);
+			const result: string = parseQueryToQueryString(query);
 
 			expect(result).toEqual(expectedResult);
 	});
 });
+
+describe('function: parseQueryToRouterString', () => {
+	it('should return string (foo) when no other attributes are set and displayString is foo', () => {
```

Comment:
Well I thought about it, but if there is only one test and we loop inside it the issue is if it fails for any one case the test is very hard to debug as in `Jasmine` it doesn't show which assertion for which item of the array failed it will only show that the test failed, that's why for maintainability and checking of tests it has to be done like this, Ugly but the best thing I guess to do